api/sys_plugins.go

```diff
@@ -23,7 +23,13 @@ type ListPluginsResponse struct {
 // ListPlugins lists all plugins in the catalog and returns their names as a
 // list of strings.
 func (c *Sys) ListPlugins(i *ListPluginsInput) (*ListPluginsResponse, error) {
-	path := fmt.Sprintf("/v1/sys/plugins/catalog/%s", i.Type)
+	path := ""
+	if i.Type == consts.PluginTypeUnknown {
```

Comment:
I think you should also handle the case where `i.Type` is not provided at all instead of having to pass `consts.PluginTypeUnknown` in order to list all plugins.

Reply:
Well, if it weren't provided, it would default to 0 which happens to also be.... `consts.PluginTypeUnknown`! :-)

Reply:
Yep, I was just reviewing your list response changes and realized that was the case!